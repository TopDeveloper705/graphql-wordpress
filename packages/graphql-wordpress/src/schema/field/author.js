import UserType from 'type/User';

export default {
  author: {
    type: UserType,
    description: 'The author object of the item.',
    resolve: (data, args, context, { rootValue: { loaders: { User } } }) =>
      data.author > 0 ? User.load(data.author) : null,
  },
};
