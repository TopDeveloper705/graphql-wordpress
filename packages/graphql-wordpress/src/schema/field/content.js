import Title from 'type/Title';
import Content from 'type/Content';
import Excerpt from 'type/Excerpt';

export const title = {
  title: {
    type: Title,
  },
};

export const content = {
  content: {
    type: Content,
  },
};

export const excerpt = {
  excerpt: {
    type: Excerpt,
  },
};
