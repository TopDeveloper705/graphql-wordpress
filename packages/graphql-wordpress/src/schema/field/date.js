import { GraphQLString } from 'graphql';

export const date = {
  date: {
    type: GraphQLString,
    description: 'The date the object was published, in the timezone of the site.',
  },
  dateGMT: {
    type: GraphQLString,
    description: 'The date the object was published, as GMT.',
  },
};

export const modified = {
  modified: {
    type: GraphQLString,
    description: 'The date the object was modified, in the timezone of the site.',
  },
  modifiedGMT: {
    type: GraphQLString,
    description: 'The date the object was modified, as GMT.',
  },
};
