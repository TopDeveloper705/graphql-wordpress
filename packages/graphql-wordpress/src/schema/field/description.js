import { GraphQLString } from 'graphql';

export default {
  description: {
    type: GraphQLString,
    description: 'The object description.',
  },
};
