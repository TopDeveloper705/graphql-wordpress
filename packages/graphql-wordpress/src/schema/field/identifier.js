import { GraphQLID, GraphQLNonNull, GraphQLString } from 'graphql';

import Guid from 'type/Guid';

export const globalIdField = () => ({
  type: new GraphQLNonNull(GraphQLID),
  description: 'Unique identifier for the object.',
  resolve: data => data.getID(),
});

export const name = {
  name: {
    type: GraphQLString,
    description: 'HTML title for the object.',
  },
};

export const id = {
  id: {
    type: GraphQLID,
    description: 'Unique identifier for the object.',
  },
};

export const slug = {
  slug: {
    type: GraphQLString,
    description: 'An alphanumeric identifier for the object unique to its type.',
  },
};

export const guid = {
  guid: {
    type: Guid,
  },
};

export const link = {
  link: {
    type: GraphQLString,
    description: 'URL to the object.',
  },
};
