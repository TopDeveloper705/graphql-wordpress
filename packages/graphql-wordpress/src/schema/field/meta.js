import { GraphQLList } from 'graphql';
import Meta from 'type/Meta';

const metaResolver = data => {
  if (!data.meta || !Object.keys(data.meta).length) {
    return null;
  }
  return Object.keys(data.meta).map(key => ({
    name: key,
    value: data.meta[key],
  }));
};

export default () => ({
  type: new GraphQLList(Meta),
  description: 'Meta fields.',
  resolve: metaResolver,
});
