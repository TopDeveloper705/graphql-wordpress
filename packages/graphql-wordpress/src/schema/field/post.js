import { GraphQLString, GraphQLBoolean } from 'graphql';

import FORMAT from 'enum/Format';

export const type = {
  type: {
    type: GraphQLString,
    description: 'Type of Post for the object.',
  },
};

export const template = {
  template: {
    type: GraphQLString,
    description: 'The theme file to use to display the object.',
  },
};

export const format = {
  format: {
    type: FORMAT,
  },
};

export const sticky = {
  sticky: {
    type: GraphQLBoolean,
    description: 'Whether or not the object should be treated as sticky.',
  },
};
