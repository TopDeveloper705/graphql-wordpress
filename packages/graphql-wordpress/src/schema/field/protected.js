import { GraphQLBoolean } from 'graphql';

export default {
  protected: {
    type: GraphQLBoolean,
    description: 'Whether the field is protected with a password.',
  },
};
