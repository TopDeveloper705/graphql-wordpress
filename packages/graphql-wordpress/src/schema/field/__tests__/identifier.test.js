import { toGlobalId } from 'graphql-relay';
import { globalIdField, name, id, slug, guid, link } from 'field/identifier';

function MockData(mockID) {
  this.id = mockID;
  this.getID = () => toGlobalId('MockData', this.id);
}

describe('Test schema type field definition', () => {
  test('Test globalIdField field', () => {
    const field = globalIdField();
    expect(field.type).toMatchSnapshot();
    const data = new MockData(867539);
    expect(field.resolve(data)).toMatchSnapshot();
  });

  test('Test name field', () => {
    const { name: field } = name;
    expect(field.type).toMatchSnapshot();
  });

  test('Test id field', () => {
    const { id: field } = id;
    expect(field.type).toMatchSnapshot();
  });

  test('Test slug field', () => {
    const { slug: field } = slug;
    expect(field.type).toMatchSnapshot();
  });

  test('Test guid field', () => {
    const { guid: field } = guid;
    expect(field.type).toMatchSnapshot();
  });

  test('Test link field', () => {
    const { link: field } = link;
    expect(field.type).toMatchSnapshot();
  });
});
