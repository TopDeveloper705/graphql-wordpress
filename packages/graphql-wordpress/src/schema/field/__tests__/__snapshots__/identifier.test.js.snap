// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test schema type field definition Test globalIdField field 1`] = `"ID!"`;

exports[`Test schema type field definition Test globalIdField field 2`] = `"TW9ja0RhdGE6ODY3NTM5"`;

exports[`Test schema type field definition Test guid field 1`] = `"Guid"`;

exports[`Test schema type field definition Test id field 1`] = `"ID"`;

exports[`Test schema type field definition Test link field 1`] = `"String"`;

exports[`Test schema type field definition Test name field 1`] = `"String"`;

exports[`Test schema type field definition Test slug field 1`] = `"String"`;
